### PR TITLE
`@defichain/salmon-wallet` with tests

### LINKED ISSUE #10: `@defichain/salmon-wallet` dogfooding implementation with testing
#### What would you like to be added:

SalmonWallet is a serverless first lightweight wallet setup that is easy to configure. Powered by DeFiCh/jellyfish ecosystem, this wallet uses Ocean APIs and required all transactions to be signed in Bech32 format.

Requiring just PrivateKey (WIF), NetworkName, and WhaleApiClient for its entirety. This wallet will construct an oracle price feed for submission, sign the oracle price feed with the given private key. Lastly, submit the price feed into the connected network.

 - [ ] Dogfooding implementation with plenty of testing
 - [ ] Use testcontainers and jellyfish-testing for testing

/assign @monstrobishi 
/triage accepted
/area core
/priority important-soon

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

`salmon-wallet` with tests

#### Which issue(s) does this PR fixes?:

Fixes #10

### CODE REVIEW

File: packages/salmon-wallet/__test__/send.test.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  }).rejects.toThrowError('SalmonWallet.send prices list is empty')
+})
+
+it('should fail on invalid oracle id', async () => {
+  const wallet = new SalmonWallet(address.privKey, 'regtest', whaleApiClient)
+
+  const prices: AssetPrice[] = [
+    {
+      token: 'BTC',
+      amount: new BigNumber(NaN),
```
Comment (fuxingloh) [inline packages/salmon-wallet/__test__/send.test.ts]:
```suggestion
      amount: new BigNumber(1),
```

File: packages/salmon-wallet/__test__/send.test.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  client = new JsonRpcClient(await container.ain.getCachedRpcUrl())
+  whaleApiClient = container.getWhaleApiClient()
+
+  await client.wallet.sendToAddress(address.address, 1)
+  await container.ain.generate(1)
+
+  oracleId = await client.oracle.appointOracle(address.address,
+    ['BTC', 'ETH', 'DOGE'].map(x => ({ token: x, currency: 'USD' })), {
+      weightage: 1.0
+    })
```
Comment (fuxingloh) [inline packages/salmon-wallet/__test__/send.test.ts]: separate this into 2 lines? easier to read that way especially when it comes to reviewing.

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: packages/salmon-wallet/__test__/send.test.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
+import waitForExpect from 'wait-for-expect'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { GenesisKeys } from '@defichain/testcontainers'
```
Comment (fuxingloh) [inline packages/salmon-wallet/__test__/send.test.ts]:
```suggestion
import { RegTestGenesisKeys } from '@defichain/jellyfish-network'
```

always avoid deprecated functions, if we keep using them, it makes it hard to migrate away from them

File: tsconfig.json [hunk 4]
```diff
@@ -5,6 +5,9 @@
     "paths": {
       "@defichain/salmon-*": [
         "packages/salmon-*/src"
+      ],
+      "@defichain/salmon": [
```
Comment (fuxingloh) [inline tsconfig.json]: to reverting this change

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 4 above)
Comment (monstrobishi) [reply tsconfig.json]: sorry can you elaborate, it looks the same as main branch

Comment (monstrobishi) [reply tsconfig.json]: ok I think i got it

Comment (monstrobishi) [reply tsconfig.json]: I managed to revert it to /main but github is still showing this change (although it says outdated here on the comments)

**Review by fuxingloh — DISMISSED** [review]

**Review by joeldavidw — DISMISSED** [review]

File: packages/salmon-wallet/__test__/send.test.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  )
+})
+
+it('should fail on empty price list', async () => {
+  const wallet = new SalmonWallet(address.privKey, 'regtest', whaleApiClient)
+
+  const prices: AssetPrice[] = []
+
+  await expect(async () => {
+    await wallet.send('invalid', prices)
```
Comment (ivan-zynesis) [inline packages/salmon-wallet/__test__/send.test.ts]:
```suggestion
    await wallet.send(oracleId, prices)
```

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Review by joeldavidw — APPROVED** [review]

**Pull request merged**